ui/src/components/LoginForm/SelectProposal.jsx

```diff
@@ -1,182 +1,123 @@
-              >
-                Reschedule
-              </Button>
-            )}
-          {session && session.is_scheduled_beamline === false && (
-            <Button
-              variant="danger"
-              className="float-end"
-              disabled // {this.state.pNumber === null}
-              onClick={this.handleSelectProposal}
-            >
-              Move here
-            </Button>
-          )}
-          <Button variant="outline-secondary" onClick={this.handleCancel}>
-            Cancel
-          </Button>
-          <Button
-            variant="primary"
-            className="float-end"
-            disabled={
-              this.state.pNumber === null ||
-              (session && session.is_scheduled_beamline === false) ||
-              session.is_scheduled_time === false
-            }
-            onClick={this.handleSelectProposal}
+  return (
+    <Modal
+      show={show}
+      backdrop={showProposalsForm || 'static'}
+      onHide={() => {
+        if (showProposalsForm) {
+          handleHide();
+        }
+      }}
+    >
+      <Modal.Header closeButton>
+        <Modal.Title>Select a session</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <Form.Control
+          type="text"
+          className="mb-3"
+          placeholder="Filter"
+          aria-label="Filter sessions"
+          value={filter}
+          onChange={(evt) => setFilter(evt.target.value)}
+        />
+
+        <Tabs id="scheduled-tab" defaultActiveKey="scheduled">
+          <Tab
+            eventKey="scheduled"
+            title={`Scheduled (${scheduledSessions.length})`}
           >
-            {this.state.proposal === null
-              ? 'Select Proposal'
-              : `Select ${this.state.proposal}`}
-          </Button>
-        </Modal.Footer>
-      </Modal>
-    );
-  }
+            <div className={styles.table}>
+              <SessionTable
+                sessions={scheduledSessions}
+                selectedSessionId={selectedSessionId}
+                onSessionSelected={setSelectedSession}
+              />
+            </div>
+          </Tab>
+          <Tab
+            eventKey="unscheduled"
+            title={`Unscheduled (${unscheduledSessions.length})`}
+          >
+            <div className={styles.table}>
+              <SessionTable
+                showBeamline
+                sessions={unscheduledSessions}
+                selectedSessionId={selectedSessionId}
+                onSessionSelected={setSelectedSession}
+              />
+            </div>
+          </Tab>
+        </Tabs>
+      </Modal.Body>
+      <Modal.Footer>
+        <ActionButton
+          selectedSession={selectedSession}
+          onClick={() => dispatch(selectProposal(selectedSessionId))}
+        />
+        <Button
+          variant="outline-secondary"
+          data-default-styles
+          onClick={() => handleHide()}
+        >
+          {showProposalsForm ? 'Cancel' : 'Sign out'}
```

Comment:
Being more explicit about what canceling does when the dialog is shown because no proposal is selected.